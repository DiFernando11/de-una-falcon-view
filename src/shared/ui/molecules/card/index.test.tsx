import { render, screen } from '@testing-library/react';
import Card from './index';
import '@testing-library/jest-dom';

describe('<Card />', () => {
  it('should render children correctly', () => {
    // Arrange
    const content = 'Contenido de la card';
    // Act
    render(<Card>{content}</Card>);
    // Assert
    expect(screen.getByText(content)).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    // Arrange
    const content = 'Snapshot Card';
    // Act
    const { asFragment } = render(<Card theme="default">{content}</Card>);
    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
