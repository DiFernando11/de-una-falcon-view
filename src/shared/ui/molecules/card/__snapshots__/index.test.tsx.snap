// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<Card /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="rounded-2xl p-6 transition-all relative overflow-hidden shadow-lg border cursor-pointer card-hover bg-white border-gray-200 text-black"
  >
    Snapshot Card
  </div>
</DocumentFragment>
`;
